### PR TITLE
Issue 247 - set default result file behaviour for sql-dump

### PR DESCRIPTION
**Pull request by Tawreh**
Original issue was #247 

Initially raised this as an issue in our internal issue tracking with CE, but it seems like a more widespread issue, so sensible to raise here instead.

There's a line in the deployment scripts that runs `drush sql-dump` and pipes the output into a bzip file:

    [jenkins@server] run: drush @site_environment sql-dump | bzip2 -f > /var/www/target_branch/db/site_environment.sql.bz2

With Drush 8, that worked fine - `sql-dump`, with no flags, printed raw SQL to the command line ready to be piped into a file.

With Drush 9, the `--result-file` flag is now required. See related issue where someone has mentioned this: https://github.com/drush-ops/drush/issues/3364

To get around this problem, we have added the following to our `drush.yml` file: 

    commands:
      sql:
        dump:
          options:
            result-file: auto

This adds `--result-file auto` every time we run `sql:dump` (including when these builds are run, and locally), ensuring we get a file dumped to our home directories. 

In summary: 

* `drush sql:dump` now requires the `--result-file` flag with a value, otherwise it errors and fails to dump a file 
* Adding `--result-file auto` saves a file and prints a message to the screen, which is picked up by your scripts and piped into a bz2 file
* Turns out that message is not valid SQL, so the script struggles to import it later on in the process

Do we remove it from our `drush.yml` and let you fix it? Do the CE scripts need to be updated so they use the result file (and clean up afterwards)? 

Pull request attached, I did try using `--gzip` flags for some things but there were so many instances where names of files changed that I just got confused and left it all as bzip2 files.

### CODE REVIEW

**Comment by Tawreh** [comment]
I did not expect this to open a new issue :thinking:  I'm so not used to Github.

File: drupal/DrupalUtils.py [hunk 1]
```diff
@@ -91,7 +91,8 @@ def get_database(shortname, branch, santise):
       dbhost = run("drush @%s_%s status  Database\ host | awk {'print $4'} | head -1" % (shortname, branch))
       run('mysqldump --single-transaction -c --opt -Q --hex-blob -u%s -p%s -h%s %s | /usr/local/bin/drupal-obfuscate.rb | bzip2 -f > ~jenkins/client-db-dumps/%s-%s_database_dump.sql.bz2' % (dbuser, dbpass, dbhost, dbname, shortname, branch))
   else:
-    run('drush @%s_%s sql-dump | bzip2 -f > ~jenkins/client-db-dumps/%s-%s_database_dump.sql.bz2' % (shortname, branch, shortname, branch))
+      dump_file = "%s_%s_database_dump.sql" % (shortname, branch)
```
Comment (gregharvey) [inline drupal/DrupalUtils.py]: Sorry to be pedantic, but these two lines are tabbed in too far and Python can be funny about that...

File: drupal/Sync.py [hunk 2]
```diff
@@ -148,7 +149,8 @@ def sync_db(orig_host, shortname, staging_shortname, staging_branch, prod_branch
           dbhost = run("drush @%s_%s status  Database\ host | awk {'print $4'} | head -1" % (shortname, prod_branch))
           run('mysqldump --single-transaction -c --opt -Q --hex-blob -u%s -p%s -h%s %s | /home/jenkins/drupal-obfuscate.rb | bzip2 -f > ~jenkins/dbbackups/drupal_%s_%s.sql.bz2' % (dbuser, dbpass, dbhost, dbname, shortname, now))
     else:
-      run('drush @%s_%s sql-dump | bzip2 -f > ~jenkins/dbbackups/drupal_%s_%s.sql.bz2' % (shortname, prod_branch, shortname, now))
+        dump_file = "drupal_%s_%s.sql" % (shortname, now)
```
Comment (gregharvey) [inline drupal/Sync.py]: Tabbing issue here too. :-)

**Comment by Tawreh** [comment]
Oh hey, I didn't get ANY notifications for this, I'm sorry - will take a look at the feedback today.

**Comment by Tawreh** [comment]
Hi all 

I've been avoiding these conflicts because I'm not totally familiar with the Drupal.py file and feel like someone with more experience is better-placed to resolve them. 

This is becoming an issue across a couple of projects now. I'm happy to try to resolve these conflicts but can't guarantee that I'll "do it right" (read: won't remove some code that's needed). 

Any pointers appreciated, any feedback on this is appreciated, especially as it's now been a month and a half! :)

**Comment by gregharvey** [comment]
Hi,

My turn to apologise, I don't get GitHub notifications either (?!!) - don't know what that's about! Anyway, I accepted Marc's PR which I think covers the same issue? Maybe we can close this one?

Greg
